feeds/scheduler/scheduler.go

```diff
@@ -0,0 +1,42 @@
+package scheduler
+
+import (
+	"time"
+
+	"github.com/ossf/package-feeds/feeds"
+	"github.com/ossf/package-feeds/feeds/crates"
+	"github.com/ossf/package-feeds/feeds/npm"
+	"github.com/ossf/package-feeds/feeds/pypi"
+	"github.com/ossf/package-feeds/feeds/rubygems"
+	log "github.com/sirupsen/logrus"
+)
+
+// ScheduledFeeds is a registry of feeds that should be run on a schedule
+var ScheduledFeeds = make(map[string]feeds.ScheduledFeed)
+
+func init() {
+	ScheduledFeeds[pypi.FeedName] = pypi.Feed{}
+	ScheduledFeeds[npm.FeedName] = npm.Feed{}
+	ScheduledFeeds[rubygems.FeedName] = rubygems.Feed{}
+	ScheduledFeeds[crates.FeedName] = crates.Feed{}
+}
+
+// PollScheduledFeeds fetches the latest packages from each registered feed
+func PollScheduledFeeds(cutoff time.Time) ([]*feeds.Package, error) {
+	packages := []*feeds.Package{}
+	for name, feed := range ScheduledFeeds {
+		logger := log.WithField("feed", name)
+		pkgs, err := feed.Latest(cutoff)
```

Comment:
What are your thoughts on running each of these in a goroutine `feed.Latest`?

Reply:
Something like this.
```
func PollScheduledFeeds(cutoff time.Time) ([]*feeds.Package, error) {
	var err error
	type success struct {
		name   string
		result []*feeds.Package
	}

	type failure struct {
		name string
		err  error
	}
	packages := []*feeds.Package{}
	results := make(chan success, len(ScheduledFeeds))
	errorsResults := make(chan failure, len(ScheduledFeeds))

	for name, feed := range ScheduledFeeds {
		go func(name string, feed feeds.ScheduledFeed) {
			pkgs, err := feed.Latest(cutoff)
			if err != nil {
				errorsResults <- failure{name, err}
			}
			results <- success{name, pkgs}
		}(name, feed)
	}

	for i := 1; i <= len(ScheduledFeeds); i++ {
		select {
		case pkgs := <-results:
			logger := log.WithField("feed", pkgs.name)
			packages = append(packages, pkgs.result...)
			logger.WithField("num_processed", len(pkgs.result)).Print("processed packages")
		case errResult := <-errorsResults:
			logger := log.WithField("feed", errResult.name)
			logger.WithError(errResult.err).Error("error fetching packages")
			err = multierror.Append(err, errResult.err)
		}
	}
	return packages, err
}

```

Reply:
I'm definitely in favor of doing this! But I think I'll leave it for a separate PR/optimization to try and keep this already complex refactor to be as simple as possible.